searchsummary/src/vespa/juniper/sumdesc.cpp

```diff
@@ -759,7 +758,7 @@ std::string SummaryDesc::get_summary(const char* buffer, size_t bytes,
                s.size(), a.charLen());
     _sumconf = NULL; // Not valid after this call.
     char_size = a.charLen();
-    return std::string(&s[0], s.size());
```

Comment:
I don't understand this could end up taking a reference to nullptr, but if size() is 0 then s[0] is certainly illegal.  Are you saying string's internal char buffer is initialized to nullptr?  Because c_str() could be called on a newly initialized string I thought the internal buffer always had at least length 1 to include the '\0'.

Reply:
Yes, it is &s[0] which is the problem. The vector's buffer is nullptr when nothing has been added, or space reserved.

Reply:
What will c_str() do then if nothing has been added?  Return a pointer to a '\0' byte stored outside the buffer or will a buffer be allocated?

Reply:
std::string has a buffer on the stack for up to 16 bytes which it uses for small strings before allocating any memory. The zero termination will be there if it is empty.